vscode/src/test/suite/testutils.ts

```diff
@@ -192,3 +192,278 @@ export async function dumpJava(): Promise<void> {
+    let localized: boolean = true;
+    const explorer: any = views.explorer;
+    const explorerLocalisableFields = ["name", "contextualTitle"];
+    let localizingKey: string;
+    var fieldVal: string;
+    var id: string;
+    for (const obj of explorer) {
+        if (!isLocalizedObj(obj, explorerLocalisableFields, obj.id, "explorer", validKeys)) localized = false;
+    }
+    return localized;
+}
+
+
+export function checkCommandsLocalisation(commands: any, validKeys: Set<string>): boolean {
+    const localisableFields = ['title'];
+    let localized: boolean = true;
+    let localizingKey: string;
+    var fieldVal: any;
+    var id: string;
+    for (const command of commands) {
+        id = command.command;
+        for (const field of localisableFields) {
+            fieldVal = command[field];
+            if (fieldVal === undefined) continue;
+            localizingKey = getlocalizingKey(fieldVal);
+            if (!isLocalizedVal(fieldVal)) {
+                localized = false;
+                console.log(`${field} of the command '${id}' is not being localized`)
+            } else if (!validKeys.has(localizingKey)) {
+                console.log(`Invalid localizing key for the command '${id}' key '${localizingKey}' not found in the package.nls.json`);
+            }
+        }
+    }
+    return localized;
+}
+
+
+/**
+ * 
+ * @param str1 
+ * @param str2 
+ * @returns Checks if the placeholders specified as {placeholder} match for str1 and str2
+ */
+function matchTemplate(str1: string, str2: string): boolean {
+    const regexp = /\{[^\{\}]*\}/g;
+    const params1 = new Set([...str1.matchAll(regexp)].map((value) => value[0]));
+    const params2 = new Set([...str2.matchAll(regexp)].map((value) => value[0]));
+    return setEqual(params1, params2);
+}
+
+function setEqual(setA: Set<string>, setB: Set<string>): boolean {
+    for (const elem of setA) {
+        if (setB.has(elem)) continue;
+        return false;
+    }
+    return setA.size === setB.size;
+}
+
+/**
+ * 
+ * @param targetString String containing placeholder in form of {placeholder}
+ * @param placeholders Set of required placeholders in the target string 
+ * @returns Whether the placeholders present in the targetString and in the placeholders set are same 
+ */
+function placeholderMatch(targetString: string, placeholders: Set<string>): boolean {
+    const regexp = /\{([^\{\}]*)\}/g;
+    const params1 = new Set([...targetString.matchAll(regexp)].map((value) => value[1]));
+    return setEqual(params1, placeholders);
+}
+
+
+
+/**
+ * 
+ * @param dictString String having key value pairs {key1:value1,key2:value2...}
+ * @returns Set of keys used in the dictString
+ */
+function getPlaceholders(dictString: string): Set<string> {
+    const placeholders = new Set<string>();
+    const cleanedDictString = dictString.replace(/"[^"]*"/g, "SOME_STRING_VALUE").replace(/'[^']*'/g, "SOME_STRING_VALUE");
+    for (const keyVal of cleanedDictString.split(',')) {
```

Comment:
Perhaps skip the first element of keyVal since that is expected to be the key?

Reply:
Hmm I don't think so, it will be a key value pair only something like `key1:value1` assuming the `cleanedDictString` is  like `"key1:value1 , key2:value2 , key3:value3..."`

Reply:
Understood. Thanks.